### ISSUE TITLE
How to I ***completely*** back off from a started git-imerge rebase?

### ISSUE THREAD

**Issue by rulatir** [mhagger/git-imerge#95]
It turns out that git-imerge abort is not a thing. Surprise! Fear!

How do I **completely** back off from a started git-imerge and return my repository to the state it was immediately before I issued the "git imerge rebase master" command?

A note to anyone who might try to convince me that there is a better and completely safe way to proceed than backing off completely: please keep in mind that I Am Not A Git Wizard. Strategies available to you thanks to your skills might not be *practically* available to me. I am 100% informedly resolved that I do in fact want to Roll It Back Completely.

**Comment by rulatir** [comment]
Allright, it's git imerge remove **followed by checking out the branch I was on before starting the merge**. I used gitk --all and still saw lots of imerge commits, but that was just due to the last of them being referenced by HEAD. After checking the original branch they are gone.

**Comment by waldyrious** [comment]
@rulatir Could you submit a PR with instructions on how to do this? I would do it myself but I'm not 100% sure I got exactly what you wrote.

**Comment by rulatir** [comment]
You mean a PR with appropriate changes to readme?

Anyways, git-imerge works by creating separate branches for its operation and commits to those. It does not touch the original branches until it is 100% done merging. Apparently `git-imerge remove` is the command to simply remove the branches created by git-imerge, but if you do it mid-merge, HEAD might remain pointing to some commit created by git-imerge even though that commit is no longer on any branch (i.e. detached HEAD). This is **very** visible if you use a visual tool that displays the commit graph (because you see the entire "fan" of git-imerge's intermediate merge commits leading up to the commit pointed to by HEAD), but once you check out your original branch (the one you were on when you called git-imerge), those commits will likely be gone from the commit graph because they will no longer be referenced by anything.

**Comment by waldyrious** [comment]
I believe I understand, thanks for the detailed explanation. I was indeed referring to an edit to the README, specifically the sequence of commands you'd recommend. Would it then be `git-imerge remove` then `git checkout <original branch>`?

I suppose this should go in the "[Suspending/resuming](https://github.com/mhagger/git-imerge#suspendingresuming)" section (whose title could have "/aborting" appended), but should that text be reworked as well?

Last time I tried using this on a huge, nasty conflict (line-ending changes across multiple files interfering with changes on those files), I eventually decided to give up, but the instructions on that section got me into an even worse (so I thought) state. I can't recall the details exactly, but I believe I tried resolving a manual merge, committing (per the instructions git-imerge gave me), then I decided to back off by doing `git merge --abort`; but then the tree was all messed up, and `git imerge continue` gave me an error message about commits not having the right parents, IIRC. Of course, this is only tangentially related, and probably my fault, so if the text there looks good and unambiguous to you, let's just add the extra paragraph about completely aborting the incremental merge.
